### PR TITLE
Golden Bike Horns now have a small delay between each flips.

### PR DESCRIPTION
**Pull request by Mercenaryblue**
:cl: 
tweak: Golden Bike Horns now permit its victims to perform a full flip before forcing them to jump again.
/:cl:

[why]: Because that was how its was intended to be. Flip non-stop, sure, but a flip nevertheless.

### CODE REVIEW

**Comment by Repukan** [comment]
Lame

File: code/game/objects/items/clown_items.dm [hunk 1]
```diff
@@ -127,20 +128,27 @@
...
 /obj/item/bikehorn/golden
 	name = "golden bike horn"
 	desc = "Golden? Clearly, it's made with bananium! Honk!"
 	icon_state = "gold_horn"
 	item_state = "gold_horn"
+	var/flip_flag = FALSE
 
 /obj/item/bikehorn/golden/attack()
-	flip_mobs()
+	if (flip_flag == FALSE)
```
Comment (Cyberboss) [inline code/game/objects/items/clown_items.dm]: if(!flip_flag)

File: code/game/objects/items/clown_items.dm [hunk 2]
```diff
@@ -127,20 +128,27 @@
...
+		flip_mobs()
 	return ..()
 
 /obj/item/bikehorn/golden/attack_self(mob/user)
-	flip_mobs()
+	if (flip_flag == FALSE)
```
Comment (Cyberboss) [inline code/game/objects/items/clown_items.dm]: if(!flip_flag)

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by banthebantz** [comment]
I see Honkbot makes its own upstream PRs now.

File: code/game/objects/items/clown_items.dm (hunk 2 above)
Comment (Mercenaryblue) [reply code/game/objects/items/clown_items.dm]: oooh, I didn't think these were automatically defined. Roger that.

**Comment by Mercenaryblue** [comment]
No, we share the same avatar. He's a bot for hippiestation.
I'm a coder for /tg/station

**Review by Cyberboss — DISMISSED** [review]

**Comment by KorPhaeron** [comment]
@Cyberboss Wouldn't it be better to do cooldown = world.time + X instead of using a timer? Or is it a negligible difference?

**Comment by lzimann** [comment]
i think we should avoid adding even more things to the timer system if we can avoid, plus doing `world.time + X` would avoid doing a proc call

**Comment by optimumtact** [comment]
prefer world.time where you can, use timers where you cant

File: code/game/objects/items/clown_items.dm [hunk 3]
```diff
@@ -127,20 +128,27 @@
 	. = ..()
 	AddComponent(/datum/component/squeak, list('sound/items/airhorn2.ogg'=1), 50)
 
+//golden bikehorn
 /obj/item/bikehorn/golden
 	name = "golden bike horn"
 	desc = "Golden? Clearly, it's made with bananium! Honk!"
 	icon_state = "gold_horn"
 	item_state = "gold_horn"
+	var/flip_flag = FALSE
```
Comment (KorPhaeron) [inline code/game/objects/items/clown_items.dm]: So you'll want to change this to var/cooldown

Comment (KorPhaeron) [inline code/game/objects/items/clown_items.dm]: So you'll want to change this to var/cooldown = 0

File: code/game/objects/items/clown_items.dm [hunk 4]
```diff
@@ -127,20 +128,27 @@
...
 /obj/item/bikehorn/golden
 	name = "golden bike horn"
 	desc = "Golden? Clearly, it's made with bananium! Honk!"
 	icon_state = "gold_horn"
 	item_state = "gold_horn"
+	var/flip_flag = FALSE
 
 /obj/item/bikehorn/golden/attack()
-	flip_mobs()
+	if (!flip_flag)
```
Comment (KorPhaeron) [inline code/game/objects/items/clown_items.dm]: And check this check to if(cooldown < world.time)

File: code/game/objects/items/clown_items.dm [hunk 5]
```diff
@@ -127,20 +128,27 @@
...
+		flip_mobs()
 	return ..()
 
 /obj/item/bikehorn/golden/attack_self(mob/user)
-	flip_mobs()
+	if (!flip_flag)
```
Comment (KorPhaeron) [inline code/game/objects/items/clown_items.dm]: And check this check to if(cooldown < world.time)

File: code/game/objects/items/clown_items.dm [hunk 6]
```diff
@@ -127,20 +128,27 @@
...
+		flip_mobs()
 	..()
 
+/obj/item/bikehorn/golden/proc/reset_flag()
```
Comment (KorPhaeron) [inline code/game/objects/items/clown_items.dm]: Delete this proc

File: code/game/objects/items/clown_items.dm [hunk 7]
```diff
@@ -149,7 +157,10 @@
 			if(istype(H.ears, /obj/item/clothing/ears/earmuffs))
 				continue
 		M.emote("flip")
+	flip_flag = TRUE
```
Comment (KorPhaeron) [inline code/game/objects/items/clown_items.dm]: Replace this line and the addtimer line with

cooldown = world.time + 7

**Review by imtakingabreakdontatme — DISMISSED** [review]

**Comment by KorPhaeron** [comment]
Actually I'll just do it for you

**Comment by duncathan** [comment]
important to note, too, that (unless it changed while I was on hiatus) timers are lower resolution than the typical `world.time` cooldowns, so they're often not ideal for very short or very precise cooldowns

**Comment by Cyberboss** [comment]
@MrStonedOne can answer that

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**